### DISCUSSION TITLE
Is there a public API for https://dandiarchive.org/search ?

### DISCUSSION THREAD

**Discussion by surchs** [dandi/dandi-archive#1859] [Q&A] [answered]
I just discovered https://dandiarchive.org/search which lets me find dandisets by searching across assets. Yet at https://api.dandiarchive.org/swagger/ I don't see any endpoints that let me query assets directly - I can only list assets themselves.

Is there a public API that lets me query assets and then returns dandisets in a similar way to how https://dandiarchive.org/search behaves?
I believe this may be related to https://github.com/dandi/dandi-archive/issues/1659

**Comment by waxlamp** [comment, marked as answer]
Hi @surchs. Sorry for the delay in replying.

Unfortunately, the search you are referring to is only available through the web application at present. This is largely because we are still figuring out how search can best work for the DANDI Archive, including questions of what search patterns are likely to be used, how to design API to accelerate those use cases, and what steps we need to take to make search both useful and performant.

As we solve these problems, we will be better positioned to offer a stable search interface through the API.

We are similarly working on a way to search through assets and generate search results related to the containing Dandisets.
